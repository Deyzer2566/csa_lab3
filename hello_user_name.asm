b =_start
b =_int_handler
ask: string "What is your name?"
hello: string "Hello, "
vosck: string "!"
name: allocate 16
name_address: word 0
name_entered: word 0
out_register: word 0xfffffc
in_register: word 0xfffffd
_start:
    and r0, 0
    or r0, ask
    bl =print

    and r0, 0
    or r0, hello
    bl =print
    
    and r0, 0
    or r0, name
loop2:
    ldr r1, [=name_entered]
    cmp r1, 0
    bz =loop2
    and r0, 0
    or r0, name
    bl =print
    and r0,0
    or r0, vosck
    bl =print
    halt
_int_handler:
    push r0
    push r1
    push r2

    ldr r0, [=in_register]
    ldr r1, [=name_address]
    cmp r1, 0
    orz r1, name
    ldr r2, [r0]
    str r2, [r1]
    cmp r2, 0
    strz r0, [=name_entered]
    or r2, -1
    str r2, [r0]
    add r1, 4
    str r1, [=name_address]

    pop r2
    pop r1
    pop r0
    exint
find_len:
    mov r1, r0
    and r0, 0
loop:
    ldr r2, [r1]
    cmp r2, 0
    bxz lr
    add r0, 1
    add r1, 4
    b =loop
print:
    push r0
    push lr
    bl =find_len
    pop lr
    mov r1, r0
    pop r0
    cmp r1, 0
    bxz lr
    ldr r2, [=out_register]
    ldr r3, [r0]
    cmp r3, 0
    str r3, [r2]
    sub r1, 1
    add r0, 4
    b =print