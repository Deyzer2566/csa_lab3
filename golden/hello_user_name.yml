in_source: |-
  b =_start
  b =_int_handler
  ask: string "What is your name?"
  hello: string "Hello, "
  vosck: string "!"
  name: allocate 16
  name_address: word 0
  name_entered: word 0
  out_register: word 0xfffffc
  in_register: word 0xfffffd
  _start:
      and r0, 0
      or r0, ask
      bl =print

      and r0, 0
      or r0, hello
      bl =print
      
      and r0, 0
      or r0, name
  loop2:
      ldr r1, [=name_entered]
      cmp r1, 0
      bz =loop2
      and r0, 0
      or r0, name
      bl =print
      and r0,0
      or r0, vosck
      bl =print
      halt
  _int_handler:
      push r0
      push r1
      push r2

      ldr r0, [=in_register]
      ldr r1, [=name_address]
      cmp r1, 0
      orz r1, name
      ldr r2, [r0]
      str r2, [r1]
      cmp r2, 0
      strz r0, [=name_entered]
      or r2, -1
      str r2, [r0]
      add r1, 4
      str r1, [=name_address]

      pop r2
      pop r1
      pop r0
      exint
  find_len:
      mov r1, r0
      and r0, 0
  loop:
      ldr r2, [r1]
      cmp r2, 0
      bxz lr
      add r0, 1
      add r1, 4
      b =loop
  print:
      push r0
      push lr
      bl =find_len
      pop lr
      mov r1, r0
      pop r0
      cmp r1, 0
      bxz lr
      ldr r2, [=out_register]
      ldr r3, [r0]
      cmp r3, 0
      str r3, [r2]
      sub r1, 1
      add r0, 4
      b =print
in_stdin: |-
  Andrey
out_code: |-
  b'0\x00\x00\xc80\x00\x01\x0c\x00\x00\x00W\x00\x00\x00h\x00\x00\x00a\x00\x00\x00t\x00\x00\x00 \x00\x00\x00i\x00\x00\x00s\x00\x00\x00 \x00\x00\x00y\x00\x00\x00o\x00\x00\x00u\x00\x00\x00r\x00\x00\x00 \x00\x00\x00n\x00\x00\x00a\x00\x00\x00m\x00\x00\x00e\x00\x00\x00?\x00\x00\x00\x00\x00\x00\x00H\x00\x00\x00e\x00\x00\x00l\x00\x00\x00l\x00\x00\x00o\x00\x00\x00,\x00\x00\x00 \x00\x00\x00\x00\x00\x00\x00!\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\xff\xff\xfc\x00\xff\xff\xfdP\x00\x00\x00X\x00\x00\x088\x00\x00\xa8P\x00\x00\x00X\x00\x00T8\x00\x00\x9cP\x00\x00\x00X\x00\x00|\x10\x02\xff\xd0\xb0\x02\x00\x001\x80\xff\xf4P\x00\x00\x00X\x00\x00|8\x00\x00|P\x00\x00\x00X\x00\x00t8\x00\x00p\xa0\x00\x00\x00 \x00\x00\x00 \x02\x00\x00 \x04\x00\x00\x10\x00\xff\xa4\x10\x02\xff\x94\xb0\x02\x00\x00Y\x82\x00|\x10\x05\x00\x00\x18\x05\x10\x00\xb0\x04\x00\x00\x19\x80\xff\x80X\x04\xff\xff\x18\x05\x00\x00p\x02\x00\x04\x18\x02\xffl(\x04\x00\x00(\x02\x00\x00(\x00\x00\x00\xb8\x00\x00\x00\x08\x03\x00\x00P\x00\x00\x00\x10\x05\x10\x00\xb0\x04\x00\x00A\x96\x00\x00p\x00\x00\x01p\x02\x00\x040\x00\xff\xe8 \x00\x00\x00 \x16\x00\x008\x00\xff\xd4(\x16\x00\x00\x08\x03\x00\x00(\x00\x00\x00\xb0\x02\x00\x00A\x96\x00\x00\x10\x04\xff \x10\x07\x00\x00\xb0\x06\x00\x00\x18\x07 \x00x\x02\x00\x01p\x00\x00\x040\x00\xff\xc4'
out_stdout: |
  What is your name?Hello, Andrey!Выполнено тактов: 42674
